From 5892a82325544ae1e7c7500432fe35d37c4d8a73 Mon Sep 17 00:00:00 2001
From: hscham <hscham@chromium.org>
Date: Wed, 11 Dec 2019 12:06:30 +0900
Subject: [PATCH] libchrome: r576279 forward compatibility patch part 7
 (flat_map).

This CL includes;
- base::flat_map alias

BUG=chromium:909719
TEST=build

Change-Id: I8746e3cad506e2524209fd69eb0f513119916177
---
 base/containers/flat_map.h | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)
 create mode 100644 base/containers/flat_map.h

diff --git a/base/containers/flat_map.h b/base/containers/flat_map.h
new file mode 100644
index 0000000..942992d
--- /dev/null
+++ b/base/containers/flat_map.h
@@ -0,0 +1,20 @@
+// Copyright (c) 2011 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This file is for forward compatibility to uprev r576279 only.
+// Do not use functions not supported by upstream flat_map.
+
+#ifndef BASE_CONTAINERS_FLAT_MAP_H_
+#define BASE_CONTAINERS_FLAT_MAP_H_
+
+namespace base {
+template <class Key,
+          class T,
+          class Hash = std::hash<Key>,
+          class Pred = std::equal_to<Key>,
+          class Alloc = std::allocator<std::pair<const Key, T>>>
+using flat_map = std::unordered_map<Key, T, Hash, Pred, Alloc>;
+}
+
+#endif // BASE_CONTAINERS_FLAT_MAP_H_
-- 
2.24.0.525.g8f36a354ae-goog

